android/src/main/java/com/swmansion/reanimated/layoutReanimation/SharedTransitionManager.java

```diff
@@ -610,4 +629,18 @@ private void enableCleaningForViewTag(int viewTag) {
       mDisableCleaningForViewTag.put(viewTag, counter - 1);
     }
   }
+
+  void orderByAnimationTypes(List<SharedElement> sharedElements) {
+    for (SharedElement sharedElement : sharedElements) {
```

Comment:
in iOS code you clear `sharedelementsWithAnimation` and the other array before this runs. Given you append to these arrays here it makes sense to have them cleared prior to adding new elements. If this happens elsewhere I'd add it here anyways